### ISSUE TITLE
device type code: "649b"

### ISSUE THREAD

**Issue by jato1569** [lprhodes/homebridge-broadlink-rm#665]
I am receiving this error at start-up.

_[Info] We've discovered an unknown Broadlink device. This likely won't cause any issues.

Please raise an issue in the GitHub repository (https://github.com/lprhodes/homebridge-broadlink-rm/issues) with details of the type of device and its device type code: "649b". The device is connected to your network with the IP address "10.0.1.xx”._

I have tried the @kiwi-cam branch, but did not fix the problem.

I would be grateful if somebody can provide any insights.

Thank you,

**Comment by kiwi-cam** [comment]
You're right. I don't have 649b. Could you let me know what device you've got and I'll get it added?

**Comment by jato1569** [comment]
It is a BroadLink rm4 Pro

**Comment by kiwi-cam** [comment]
homebridge-broadlink-rm-pro@4.2.8 will support 649b now. Note there are some issues outstanding with the RM4:
- Only IR Learning works currently (RF doesn't). [You can use the python cli to get around this](https://github.com/kiwi-cam/homebridge-broadlink-rm/issues/10#issuecomment-655197879).
- The new in cable Temperature sensor isn't supported (I've added code to the BETA but hasn't been able to test)

**Comment by Jahjahshaka** [comment]
same issue with device type code: "653a". RM4 mini. any help?

**Comment by Jahjahshaka** [comment]
I'm making little progress here...I managed to get the device showing: 

Discovered Manual RM Pro Device (2223) at 192.168.31.17 (a0::4:3::b0::1:2::54::9:d)

But I have a rm4 mini, and using another plugin will show the same discovery message with rm3 pro...
and the real issue I have is the learning process. nothing happen when I press any button and it'd time out. 
I've been checking many post with similar issues, but I'm running out of option.

**Comment by kiwi-cam** [comment]
You won't be able to use an RM4 device with this plugin. Changes were made to the communications required and this one isn't being maintained anymore.

You'll need to use [homebridge-broadlink-rm-pro](https://github.com/kiwi-cam/homebridge-broadlink-rm). Adding a hosts section will work and I'll get the 653a device added in the next version.

**Comment by Jahjahshaka** [comment]
Thanks for your help! plugin wise, i actually tried them all...and I got lost between all the posts, but so far I'm actually using homebridge-broadlink-rm-pro, since it's the one that was latest updated. I'll wait for the next version then, thanks again,you're doing an amazing job!

**Comment by Jahjahshaka** [comment]
New update fixed it all! I can now use the learn button and get what I need. Big thanks to you, my problem is solved!

**Comment by Jahjahshaka** [comment]
Sooo....I managed to set up my AC, I can turn it on and change the temperature, which is great. there is a little something that kinda bother me a bit though. Here the thing: every time I ask Siri to change the temperature, he says: "I changed it from 0 degree to 25 degree" and I can tell it's linked to the fact that I have no temperature sensor to pick this info from, so it's always 0. is there anyway to just have the same number showing on top (see attached picture)? I might get a sensor later on, but in the meantime, I'm curious to know if a work around could do it. I tried "pseudoDeviceTemperature" to get a decent number and not 0, but homekit would freeze with it. ( showing the right number though, but the app would crash) I've seen that I could use a file to pick the number, but got no success with it either. Last would be MQTT, which i have no knowledge about, and anyway, for what i understood, it needs to get the info from a sensor, which I don't have (yet). Any help would be much appreciated.
![Screen Shot 2020-12-19 at 11 20 56 AM](https://user-images.githubusercontent.com/24300894/102679676-550dff00-41ec-11eb-9d85-3764707d9547.png)
